newIDE/app/src/MainFrame/index.js

```diff
@@ -254,6 +254,19 @@ const MainFrame = (props: Props) => {
             performance.now(),
           ]);
           console.info('Startup times:', getStartupTimesSummary());
+
+          const {
+            getAutoOpenMostRecent,
+            getRecentProjectFiles,
+            getProjectHasOpened,
+          } = preferences;
+
+          if (
+            getAutoOpenMostRecent() &&
+            getProjectHasOpened() &&
+            getRecentProjectFiles()[0]
+          )
+            openFromFileMetadataWithStorageProvider(getRecentProjectFiles()[0]);
```

Comment:
There are three checks here,
1. If this is enabled by the user's preferences or not.
2. If the project was closed before closing the app or not.
3. If there exists a recent file or not.

This should in most cases stop the unwanted opening of projects I guess.

Reply:
This looks correct!